### PR TITLE
RAS-1253 Migrate Fronstage from GCR to GAR

### PR DESCRIPTION
**Pull request by arroyoAle**
# What and why?
Migrates frontstage to GAR
# How to test?
- Build out your env
- deploy frontstage using this PR and `migrate-frontstage-to-GAR` config branch
- update the image location in the helm deployment to point to the GAR - currently pointing at the management one to prevent issues with other PRs
  - europe-west2-docker.pkg.dev/`management project id`/images 
# Jira
[RAS-1253](https://jira.ons.gov.uk/browse/RAS-1253)

### CODE REVIEW

File: _infra/helm/frontstage/values.yaml [hunk 1]
```diff
@@ -3,8 +3,8 @@ namespace: minikube
 armorPolicy: "ras-cloud-armor-policy"
 
 image:
-  devRepo: eu.gcr.io/ons-rasrmbs-management
-  name: eu.gcr.io/ons-rasrmbs-management
+  devRepo: europe-west2-docker.pkg.dev/ons-rasrmbs-management/images
+  name: europe-west2-docker.pkg.dev/ons-rasrmbs-management/images
```
Comment (arroyoAle) [inline _infra/helm/frontstage/values.yaml:7]: Will be updated to the ci project before merging

**Comment by pricem14pc** [comment]
Not progressing as discussed

File: _infra/helm/frontstage/values.yaml (hunk 1 above)
Comment (arroyoAle) [reply _infra/helm/frontstage/values.yaml:7]: This is to prevent issues with other cards being progressed

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -16,8 +16,9 @@ on:
 
 env:
   IMAGE: frontstage
-  REGISTRY_HOSTNAME: eu.gcr.io
+  REGISTRY_HOSTNAME: europe-west2-docker.pkg.dev
   HOST: ${{ secrets.GOOGLE_PROJECT_ID }}
+  GAR_REPOSITORY: ${{ secrets.GAR_REPOSITORY }}
```
Comment (pricem14pc) [inline .github/workflows/main.yml:21]: I appreciate we've copied the previous migrations, but unless `secret.GAR_REPOSITORY` will resolve to different values per repo we should probably not make this a 'secret'. I think we should make it `images` explicitly in the env: vars and do that for all upcoming migrations too.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -16,8 +16,9 @@ on:
...
   RELEASE_HOST: ${{ secrets.RELEASE_PROJECT_ID }}
```
Comment (pricem14pc) [inline .github/workflows/main.yml:22]: I'm finding these confusing and think we should take the opportunity to rationalise and rename.

The `HOST` and `RELEASE_HOST` are incorrectly named and misleading. We should rename `RELEASE_HOST` to `GAR_GOOGLE_PROJECT_ID` and rename `HOST` to `SPINNAKER_GOOGLE_PROJECT_ID`.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -84,11 +85,11 @@ jobs:
       - name: Build Docker Image
         if: github.ref != 'refs/heads/main'
         run: |
-          docker build -t "$REGISTRY_HOSTNAME"/"$HOST"/"$IMAGE":${{ env.pr_number }} -f _infra/docker/Dockerfile .
+          docker build -t "$REGISTRY_HOSTNAME"/"$RELEASE_HOST"/"$GAR_REPOSITORY"/"$IMAGE":${{ env.pr_number }} -f _infra/docker/Dockerfile .
```
Comment (pricem14pc) [inline .github/workflows/main.yml:88]:
```suggestion
          docker build -t "$REGISTRY_HOSTNAME"/"$GAR_GOOGLE_PROJECT_ID"/images/"$IMAGE":${{ env.pr_number }} -f _infra/docker/Dockerfile .
```

Comment (pricem14pc) [reply .github/workflows/main.yml:88]: The above is an example of the path based on my suggestions and comments above

**Review by pricem14pc — CHANGES_REQUESTED** [review]
Suggestions to be discussed

File: .github/workflows/main.yml (hunk 2 above)
Comment (arroyoAle) [reply .github/workflows/main.yml]: Good point, I've updated it

File: .github/workflows/main.yml (hunk 3 above)
Comment (arroyoAle) [reply .github/workflows/main.yml]: Yeah thats true, I'll update them for this one and add it to the next card

File: .github/workflows/main.yml (hunk 4 above)
Comment (arroyoAle) [reply .github/workflows/main.yml]: As above

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -16,9 +16,10 @@ on:
 
 env:
   IMAGE: frontstage
-  REGISTRY_HOSTNAME: eu.gcr.io
-  HOST: ${{ secrets.GOOGLE_PROJECT_ID }}
-  RELEASE_HOST: ${{ secrets.RELEASE_PROJECT_ID }}
+  REGISTRY_HOSTNAME: europe-west2-docker.pkg.dev
+  SPINNAKER_GOOGLE_PROJECT_ID: ${{ secrets.GOOGLE_PROJECT_ID }}
```
Comment (pricem14pc) [inline .github/workflows/main.yml:20]:
```suggestion
  SPINNAKER_GOOGLE_PROJECT_ID: ${{ secrets.SPINNAKER_GOOGLE_PROJECT_ID }}
```

Comment (pricem14pc) [reply .github/workflows/main.yml:20]: I have added the above secret as I think this should be consistent throughout. We should remove the previous secret once this PR is merged.

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -16,9 +16,10 @@ on:
...
+  GAR_REPOSITORY: images
+  GAR_GOOGLE_PROJECT_ID: ${{ secrets.RELEASE_PROJECT_ID }}
```
Comment (pricem14pc) [inline .github/workflows/main.yml:22]:
```suggestion
  GAR_GOOGLE_PROJECT_ID: ${{ secrets.GAR_GOOGLE_PROJECT_ID }}
```

Comment (pricem14pc) [reply .github/workflows/main.yml:22]: I have added the above secret as I think this should be consistent throughout. We should remove the previous secret once this PR is merged.

File: .github/workflows/main.yml (hunk 3 above)
Comment (pricem14pc) [reply .github/workflows/main.yml]: Looks much better, the vars now accurately reflect their purpose.

File: .github/workflows/main.yml (hunk 4 above)
Comment (pricem14pc) [reply .github/workflows/main.yml]: Happy to leave `GAR_REPOSITORY` as a var, and that its value is now explicit rather than a GHA secret.

**Review by pricem14pc — APPROVED** [review]
Happy with the changes. Reminder to revert the project references.

**Review by matthew-robinson-ons — APPROVED** [review]
Tested and all working. Looked over the other updates and all make sense.

**Pull request merged**